### ISSUE TITLE
Fatal error in sharing inertia data for multi-guard user management system 

### ISSUE THREAD

**Issue by luiyongsheng** [joelbutcher/socialstream#184]
**Describe the bug**
I'm building an internal tool that have `User` and `Admin` which both are separated `Illuminate\Foundation\Auth\User as Authenticatable` models. However when `Admin` successfully logged in, the `Admin` model which don't have `connectedAccounts` relationship broke the `ShareInertiaData` middleware.

**Screenshots**
<img width="1387" alt="image" src="https://user-images.githubusercontent.com/13586871/160941694-829d56b1-7804-403c-869a-7012c700dfcb.png">

**Environment context**
- Socialstream version: 3.4.1
- Jetstream stack: Inertia
- Laravel version: 9.6.0
- PHP version: 8.1.4

**Comment by luiyongsheng** [comment]
Temporary workaround

```
    public function connectedAccounts(): HasMany
    {
        return $this->hasMany(ConnectedAccount::class, 'user_id', 'id');
    }
```

and also remove the `User` type inside `app\Policies\TeamPolicy.php` if needed:

```
    public function create($user)
    {
        return true;
    }
```

**Comment by joelbutcher** [comment]
Hi there, thanks for using Socialstream! 

This package is designed, like Laravel Jetstream, to work with a single Authenticatable user model. The work around above solves the issue for your use case. I am therefore closing this issue as, unfortunately, I cannot cater for all use cases and possible usages of this package.

However, if you strongly feel that a change is required here, please send in a PR. Alternatively, you are free to fork this package and maintain a version more tailored to your projects needs.

**Comment by joelbutcher** [comment]
Thanks! Released in 3.4.2
